src/bech32.c

```diff
@@ -206,22 +218,26 @@ int wally_addr_segwit_from_bytes(const unsigned char *bytes, size_t bytes_len,
     char result[90];
     size_t push_size;
     int ret;
+    uint8_t witver;
 
     if (output)
         *output = 0;
 
     if (!addr_family || flags || !bytes || !bytes_len || !output)
         return WALLY_EINVAL;
 
-    if (bytes[0] != 0)
-        return WALLY_EINVAL; /* Only v0 witness programs are currently allowed */
+    witver = bytes[0];
+    if (witver != OP_0 && witver != OP_1)
+        return WALLY_EINVAL;
+        
+    witver &= 0x0F; /* Convert OP_0 -> 0, OP_1 -> 1 etc */
 
     ret = script_get_push_size_from_bytes(bytes + 1, bytes_len - 1, &push_size);
-    if (ret != WALLY_OK || (push_size != HASH160_LEN && push_size != SHA256_LEN))
+    if (ret != WALLY_OK || (witver == 0 && push_size != HASH160_LEN && push_size != SHA256_LEN))
```

Comment:
For Taproot you can add this constraint (maybe move `ret != WALLY_OK` to a separate check):
```c
if (ret != WALLY_OK || (witver == 1 && push_size != SHA256_LEN))
```